src/asynk/async_queue.rs

```diff
@@ -54,12 +60,40 @@ where
     }
 
     pub async fn insert_task(&mut self, task: &dyn AsyncRunnable) -> Result<u64, AsyncQueueError> {
-        let json_task = serde_json::to_value(task).unwrap();
+        let metadata = serde_json::to_value(task).unwrap();
         let task_type = task.task_type();
 
-        self.execute_one(INSERT_TASK_QUERY, &[&json_task, &task_type])
+        self.execute_one(INSERT_TASK_QUERY, &[&metadata, &task_type])
             .await
     }
+    pub async fn update_task_state(
+        &mut self,
+        task: &Task,
+        state: &str,
+    ) -> Result<u64, AsyncQueueError> {
+        self.execute_one(
+            UPDATE_TASK_STATE_QUERY,
+            &[&state, &task.updated_at, &task.id],
```

Comment:
You're passing the same `updated_at` timestamp from the task struct. I think we get the current time and pass it as `updated_at`

```rust
updated_at = Utc::now()
&[&state, updated_at, &task.id]
```

Reply:
@ayrat555 i will correct that